external-crates/move/move-compiler/src/parser/ast.rs

```diff
@@ -103,6 +103,7 @@ pub struct Script {
 #[derive(Debug, PartialEq, Clone, Eq)]
 pub enum Use {
     Module(ModuleIdent, Option<ModuleName>),
+    Modules(Vec<(Use, Loc)>), // Loc tracks end of inner form
```

Comment:
I feel like a lot of the location information is hacked together as a result of some strangeness here. I might recommend the following

```
pub enum Use {
    ModuleUse(ModuleIdent, ModuleUse),
    NestedModuleUses(LeadingNameAccess, Vec<(ModuleName, ModuleUse)>),
    Fun {
        visibility: Visibility,
        function: Box<NameAccessChain>,
        ty: Box<NameAccessChain>,
        method: Name,
    },
}

pub enum ModuleUse {
    Module(Option<ModuleName>),
    Members(Vec<(Name, Option<Name>)>),
}
```

Reply:
I revised it to this :) I'm a large fan of destructuring as part of parsing, but I appreciate that it might make it harder to autoformat later.